src/bvals/bvals.hpp

```diff
@@ -181,7 +181,6 @@ class BoundaryValues : public BoundaryBase, // public BoundaryPhysics,
   // non-inhertied / unique functions (do not exist in BoundaryVariable objects):
   // (these typically involve a coupled interaction of boundary variable/quantities)
   // ------
-  void RestrictBoundaries();
```

Comment:
Boundaries are now restricted as part of pack-in-one `SendBoundaryBuffers` function. The only place where `RestrictBoundaries` was called before is in the code path using the non-in-one machinery, which this PR removes, so that's why I'm also removing `RestrictBoundaries`, since it's no longer used.

And yes, the advection outflow test still works. I've also added using outflow and reflecting boundaries to the sparse advection test.